### PR TITLE
Add Mattermost desktop profile

### PR DESCRIPTION
**Pull request by onovy**

### CODE REVIEW

File: etc/profile-m-z/mattermost-desktop.profile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+include globals.local
+
+noblacklist ${HOME}/.config/Mattermost
+
+include disable-common.inc
+include disable-devel.inc
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-programs.inc
+include disable-passwdmgr.inc
```
Comment (rusty-snake) [inline etc/profile-m-z/mattermost-desktop.profile:15]: `include disable-shell.inc` (if it works) ? 
`include disable-xdg.inc`?

File: etc/profile-m-z/mattermost-desktop.profile [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Firejail profile for mattermost-desktop
+# This file is overwritten after every install/update
+# Persistent local customizations
+include mattermost-desktop.local
+# Persistent global definitions
+include globals.local
+
+noblacklist ${HOME}/.config/Mattermost
```
Comment (rusty-snake) [inline etc/profile-m-z/mattermost-desktop.profile:8]: Add `blacklist ${HOME}/.config/Mattermost` to disable-programs.inc.

File: etc/profile-m-z/mattermost-desktop.profile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-programs.inc
+include disable-passwdmgr.inc
+
+mkdir ${HOME}/.config/Mattermost
+whitelist ${DOWNLOADS}
+whitelist ${HOME}/.config/Mattermost
+include whitelist-common.inc
+include whitelist-var-common.inc
```
Comment (rusty-snake) [inline etc/profile-m-z/mattermost-desktop.profile:21]: `include whitelist-runuser-common.inc`?
`include whitelist-usr-share-common.inc`?

File: etc/profile-m-z/mattermost-desktop.profile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+caps.drop all
+netfilter
+nodvd
+nogroups
+notv
+nou2f
+novideo
+noroot
+shell none
+protocol unix,inet,inet6
```
Comment (rusty-snake) [inline etc/profile-m-z/mattermost-desktop.profile:32]: `tracelog` (if it works)

File: etc/profile-m-z/mattermost-desktop.profile [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+mkdir ${HOME}/.config/Mattermost
+whitelist ${DOWNLOADS}
+whitelist ${HOME}/.config/Mattermost
+include whitelist-common.inc
+include whitelist-var-common.inc
+
+caps.drop all
+netfilter
+nodvd
+nogroups
```
Comment (rusty-snake) [inline etc/profile-m-z/mattermost-desktop.profile:26]: `nonewprivs`

File: etc/profile-m-z/mattermost-desktop.profile [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+notv
+nou2f
+novideo
+noroot
```
Comment (rusty-snake) [inline etc/profile-m-z/mattermost-desktop.profile:30]: Move up (below `nogroups`/`nonewprivs`).

File: etc/profile-m-z/mattermost-desktop.profile [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+shell none
+protocol unix,inet,inet6
+
+disable-mnt
+private-dev
+private-tmp
+private-etc machine-id,resolv.conf,fonts
```
Comment (rusty-snake) [inline etc/profile-m-z/mattermost-desktop.profile:37]: - `ca-certificates,ssl,pki,crypto-policies,nsswitch.conf`??
- sort alphabetical.
- Move above `private-tmp`

-----

`dbus-user none`?
`dbus-system none`

File: etc/profile-m-z/mattermost-desktop.profile [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+nogroups
+notv
+nou2f
+novideo
+noroot
+shell none
+protocol unix,inet,inet6
+
+disable-mnt
+private-dev
```
Comment (rusty-snake) [inline etc/profile-m-z/mattermost-desktop.profile:35]: `private-cache`?

File: etc/profile-m-z/mattermost-desktop.profile (hunk 4 above)
Comment (onovy) [reply etc/profile-m-z/mattermost-desktop.profile:32]: it doesn't

File: etc/profile-m-z/mattermost-desktop.profile (hunk 5 above)
Comment (onovy) [reply etc/profile-m-z/mattermost-desktop.profile:30]: too old firejail :(

File: etc/profile-m-z/mattermost-desktop.profile (hunk 7 above)
Comment (onovy) [reply etc/profile-m-z/mattermost-desktop.profile]: too old firejail :(

File: etc/profile-m-z/mattermost-desktop.profile (hunk 5 above)
Comment (rusty-snake) [reply etc/profile-m-z/mattermost-desktop.profile:30]: ????

File: etc/profile-m-z/mattermost-desktop.profile (hunk 7 above)
Comment (rusty-snake) [reply etc/profile-m-z/mattermost-desktop.profile]: You can test with `blacklist /run/dbus` (dbus-system none) and `blacklist ${RUNUSER}/bus`.

The flatpak has the following policy, so (native) notifications will be broken.
```
dbus-user filter
dbus-user.own com.mattermost.Desktop
dbus-user.talk org.freedesktop.Notifications
dbus-system none
```

I would say add the dbus-user rules commented.

File: etc/profile-m-z/mattermost-desktop.profile (hunk 6 above)
Comment (rusty-snake) [reply etc/profile-m-z/mattermost-desktop.profile]: Its missing in the latest commit. Forgotten or broken?

Do `seccomp` and `protocol` break?

Comment (onovy) [reply etc/profile-m-z/mattermost-desktop.profile]: Mattermost is Electron-based app so I removed them as you recommended in Signal MR

Comment (rusty-snake) [reply etc/profile-m-z/mattermost-desktop.profile]: +1, then the nonewprivs and caps option should also be removed/changed like the command in #2946.

File: etc/profile-m-z/mattermost-desktop.profile [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+notv
+nou2f
+novideo
+shell none
+
+disable-mnt
+private-cache
+private-dev
+private-tmp
+private-etc alternatives,ca-certificates,crypto-policies,fonts,ld.so.cache,ld.so.conf,ld.so.conf.d,ld.so.preload,machine-id,nsswitch.conf,pki,resolv.conf,ssl
```
Comment (rusty-snake) [inline etc/profile-m-z/mattermost-desktop.profile:40]: flip

**Comment by rusty-snake** [comment]
Merged, Thanks.

**Pull request merged**